packages/plugin-ext/src/common/proxy-handler.ts

```diff
@@ -50,11 +59,13 @@ export class ClientProxyHandler<T extends object> implements ProxyHandler<T> {
     private initializeRpc(): void {
         // we need to set the flag to true before waiting for the channel provider. Otherwise `get` might
         // get called again and we'll try to open a channel more than once
+        this.initCallback.reportInit(this.id);
         this.isRpcInitialized = true;
         const clientOptions: RpcProtocolOptions = { encoder: this.encoder, decoder: this.decoder, mode: 'clientOnly' };
         this.channelProvider().then(channel => {
             const rpc = new RpcProtocol(channel, undefined, clientOptions);
             this.rpcDeferred.resolve(rpc);
+            this.initCallback.reportInitDone(this.id);
```

Comment:
I'm wondering whether we could re-use the rpcDeferred in our synchronization class, something like `initializeProxy(id, deferred)` and have the implementation resolve it and remove it from the queue when deferred is resolved. Just a thought though.